src/google/protobuf/util/field_mask_util_test.cc

```diff
@@ -747,6 +749,36 @@ TEST(FieldMaskUtilTest, TrimMessage) {
   EXPECT_EQ(11, oneof_msg.oneof_uint32());
 }
 
+TEST(OneofMergeMaskTest, OneofMergeMaskTests) {
+  TestOneOfMerging demo;
+  demo.set_id(100500);
+  demo.set_foo("foo_value");
+
+  FieldMask mask;
+  mask.add_paths("bar.baz");
+
+  TestOneOfMerging dest1;
+  google::protobuf::util::FieldMaskUtil::MergeMessageTo(demo, mask, {}, &dest1);
+
+  EXPECT_EQ(dest1.demo_case(), TestOneOfMerging::DEMO_NOT_SET);
+  EXPECT_NE(dest1.demo_case(), TestOneOfMerging::kBar);
+  EXPECT_FALSE(dest1.has_bar());
+  EXPECT_FALSE(dest1.has_foo());
+
+  OneOfSubMsg* sub_msg = new OneOfSubMsg();
```

Comment:
These three lines can be simplified to just one line: `demo.mutable_bar()->set_baz("baz_value");`